### ISSUE TITLE
`Object.getCenterPoint` relative to what?

### ISSUE THREAD

**Issue by ShaMan123** [fabricjs/fabric.js#7717]
`getCenterPoint` is another method that is wrong across fabric because in some methods the center point should be in canvas coordinate plane and in others in the parent plane.
What do you prefer?
```
getCenterPoint(relative?)
// or
getCenterPoint(absolute?)
```

```
getCenterPoint()
getCenterPointRelativeToParent() // which will be the same if object in directly on canvas
```

**Comment by melchiar** [comment]
I feel like `getCenterPoint()` method should always be relative to the canvas. Having a separate `getCenterPointRelativeToParent()` method makes more sense to me than adding a `relative | absolute` argument to the current method since not every object is part of a group. `getCenterPointRelativeToParent()` can therefore just return null when not appropriate, while `getCenterPoint()` will always have a value.

**Comment by ShaMan123** [comment]
I like that

**Comment by ShaMan123** [comment]
Rethinking... canvas is a parent of the object so shouldn't it return a point from `getCenterPointRelativeToParent()`?

**Comment by ShaMan123** [comment]
This is a **MAJOR** issue

**Comment by ShaMan123** [comment]
The group rewrite makes `object_origin.mixin` inadequate. Everything there was written in thought of a simple tree of object under canvas without nesting

**Comment by melchiar** [comment]
True, canvas is an object's parent, but if we have `getCenterPoint()` to get the canvas relative center, I see no reason to duplicate that functionality when calling `getCenterPointRelativeToParent()` on ungrouped objects. If necessary, an app could always call `obj.getCenterPointRelativeToParent() || obj.getCenterPoint()` though I'm not sure what use that would be.

Of course, allowed nested groups will complicate things even further...

**Comment by ShaMan123** [comment]
Take a look at [7643132](https://github.com/fabricjs/fabric.js/commit/7643132b90ef57a395f46e504ab64bbbb572b40d) 
What do you think?
Instead of writing `obj.getCenterPointRelativeToParent() || obj.getCenterPoint()` I exposed `getRelativeCenterPoint` which is needed for internals

**Comment by ShaMan123** [comment]
In naming it is much better and less confusing

**Comment by Gerwin-prog** [comment]
Shouldn't GetCenterPoint() always return the center of the object it's called upon?

**Comment by ShaMan123** [comment]
> Shouldn't GetCenterPoint() always return the center of the object it's called upon?

It depends from where you are measuring. In other words what do mean when you point to (0,0)

**Comment by Gerwin-prog** [comment]
> > Shouldn't GetCenterPoint() always return the center of the object it's called upon?
> 
> It depends from where you are measuring. In other words, what do mean when you point to (0,0)

Can you give me an example of when it _**shouldn't**_ return it's own centerpoint?

**Comment by ShaMan123** [comment]
Always should it return it's center point.
Currently however, it returns the center point relative to it's direct parent.
Let's say you are sitting in the canvas and calling `getCenterPoint` and it returns (50,50). This means the object is translated (50,50) from canvas top left corner.
Now let's say you are sitting in a group and call `getCenterPoint` on the same object. If it returns (50,50) the fabric will be convinced that the object is translated (50,50) from group's center.
This is very wrong of course.

**Comment by ShaMan123** [comment]
There are circumstances where you need a point in a relative plane

**Comment by Gerwin-prog** [comment]
Thanks for the explanation, I agree that the 2nd scenario seems wrong, and that sometimes the point to the relative plane are required

**Comment by ShaMan123** [comment]
fixed by #7726
